lntest/itest/lnd_test.go

```diff
@@ -13928,6 +13928,93 @@ func testHoldInvoicePersistence(net *lntest.NetworkHarness, t *harnessTest) {
 	}
 }
 
+// respondChannelAccept receives a *lnrpc.ChannelAcceptRequest from the
+// funding manager and responds with a predicate telling the funding manager
+// to accept or reject the open channel request.
+func respondChannelAccept(predStream lnrpc.Lightning_ChannelAcceptorClient,
+	response *lnrpc.ChannelAcceptResponse, t *harnessTest) {
+	go func() {
+		// Opening the channel should send a *lnrpc.ChannelAcceptRequest to the stream.
+		req, err := predStream.Recv()
+		if err != nil {
+			t.Fatalf("unable to receive open channel request: %v", err)
+		}
+
+		// If a response was passed, we send it to the stream.
+		if response != nil {
+			response.PendingChanId = req.PendingChanId
+			if err := predStream.Send(response); err != nil {
+				t.Fatalf("unable to accept open channel request: %v", err)
+			}
+		}
+	}()
+}
+
+func testPredicateChannelAccept(net *lntest.NetworkHarness, t *harnessTest) {
+	ctxb := context.Background()
+
+	ctxt, cancel := context.WithCancel(ctxb)
+	defer cancel()
+
+	// Open up a bidirectional stream so that Bob can respond to incoming
+	// open channel requests.
+	bobPredStream, err := net.Bob.ChannelAcceptor(ctxt)
+	if err != nil {
+		t.Fatalf("unable to create channel acceptor stream for bob: %v", err)
+	}
+
+	// Accept the channel request.
+	acceptReponse := &lnrpc.ChannelAcceptResponse{
```

Comment:
is the pending chanid on all these responses zero?

Reply:
Nope it's set in the `respondChannelAccept` method